staging/src/k8s.io/apiserver/pkg/authentication/request/headerrequest/requestheader_test.go

```diff
@@ -152,6 +152,35 @@ func TestRequestHeader(t *testing.T) {
 			},
 			expectedOk: true,
 		},
+
+		"escaped extra keys": {
+			nameHeaders:        []string{"X-Remote-User"},
+			groupHeaders:       []string{"X-Remote-Group"},
+			extraPrefixHeaders: []string{"X-Remote-Extra-"},
+			requestHeaders: http.Header{
+				"X-Remote-User":                                            {"Bob"},
```

Comment:
can you add a test specifically with the `+` character, to make sure that round trips correctly?

Reply:
added one in the roundtripper test, but I can do so here, too

Reply:
done